kernels_utils.py

```diff
@@ -0,0 +1,160 @@
+    parser.add_argument('--seed', default=42, type=int,
+                        help="manual random seed")
+
+    # Inject workload specific options
+    if workload_args is not None:
+        workload_args(parser)
+
+    # Parse arguments
+    args = parser.parse_args()
+
+    # Run full data by default
+    if (not args.full_data) and (not args.reduced_data):
+        args.full_data = True
+
+    # Set random number seeds
+    torch.manual_seed(args.seed)
+    np.random.seed(args.seed + 1)
+    random.seed(args.seed + 2)
+
+    # Dump configs
+    if args.verbose > 0:
+        print(args)
+
+    return args
+
+# -------------------------------------------------------------------------
+# Key kernel wrapper class
+# -------------------------------------------------------------------------
+
+
+class KeyKernel:
+
+    def __init__(self, name, kernel_impl, full_data, reduced_data):
+        self.name = name
+        self.kernel_impl = kernel_impl
+        self.full_data = full_data
+        self.reduced_data = reduced_data
+
+    def run_full_data(self, hammerblade):
+        self.kernel_impl(self.full_data, hammerblade)
+
+    def run_reduced_data(self, hammerblade):
+        self.kernel_impl(self.reduced_data, hammerblade)
+
+    def _print(self):
+        print("kernel %s:" % (self.name))
+        print("  - full data    %s" % (str(self.full_data)))
+        print("  - reduced data %s" % (str(self.reduced_data)))
+        print()
+
+
+# -------------------------------------------------------------------------
+# Key kernel impl
+# -------------------------------------------------------------------------
+
+# ------- addmm ----------
+def addmm(inputs, hammerblade):
+    # Unpack inputs
+    M = inputs[0]
+    N = inputs[1]
+    P = inputs[2]
+
+    # Create random inputs
+    self = torch.randn(M, P)
+    mat1 = torch.randn(M, N)
+    mat2 = torch.randn(N, P)
+
+    # Move to HB if necessary
+    if hammerblade:
+        self = self.hammerblade()
+        mat1 = mat1.hammerblade()
+        mat2 = mat2.hammerblade()
+
+    print("  --- running addmm (%d, %d, %d) ---" % (M, N, P))
+
+    # Timer
+    start = time.time()
+
+    torch.addmm(self, mat1, mat2)
+
+    end = time.time()
+
+    print("  --- %s seconds ---" % (end - start))
+
+
+# ------- add ----------
+def add(inputs, hammerblade):
+    # Unpack inputs
+    N = inputs[0]
+
+    # Create random inputs
+    data = torch.randn(N)
+
+    # Move to HB if necessary
+    if hammerblade:
+        data = data.hammerblade()
+
+    print("  --- running add (%d) ---" % (N))
+
+    # Timer
+    start = time.time()
+
+    torch.add(data, data)
+
+    end = time.time()
+
+    print("  --- %s seconds ---" % (end - start))
+
+# -------------------------------------------------------------------------
+# Create app kernel wrapper
+# -------------------------------------------------------------------------
+
+
+def create_kernel_wrapper(key_kernels):
```

Comment:
This seems to be a key function in the api? A small doc string explaining the usage might help...